packages/components/molecules/f-rating/src/components/Rating.vue

```diff
@@ -80,18 +100,17 @@ export default {
                     rating: this.starRating,
                     total: this.maxStarRating
                 });
-        }
-    },
+        },
 
-    methods: {
         /**
-         * Check `star` against value passed by consumer to allow empty stars to render.
+         * Calculate a percentage from the `starRating` value passed in by the consuming application.
          *
-         * @param star {Number}
-         * @returns {boolean}
+         * @returns {string}
          */
-        isRatingStarFilled (star) {
-            return star <= this.starRating;
+        getRatingStarPercentage () {
+            const percentage = (this.starRating / this.maxStarRating) * 100;
+
+            return `width: ${percentage}%`;
```

Comment:
we could keep the logic here but instead of inlining a width declaration, set the width value as a css variable in the style declaration above i.e. `--starRatingPercentage`. 

The variable will be scoped to the element you add the inline style declaration to and it _might_ be a little more maintenance friendly in the future as you can just set (as an example):

```
.c-rating-mask {
  width: var(--starRatingPercentage);
}
```

The benefit is the actual width rule stays in the css. We're just changing a variable value in the template which (in my opinion) is a bit neater :)